### ISSUE TITLE
Callbacks silently fail if callback isn't known (or implemented?)

### ISSUE THREAD

**Issue by dogweather** [guard/guard#903]
This silent failure makes callbacks very difficult to debug. With Guard/RSpec, this works:

```ruby
  callback(:start_end) {
    `...`
  }
```

...but these seem to have no effect and give no error output:

```ruby
  callback(:run_all_end) {
    `...`
  }
```

```ruby
  callback(:xxxxxxx) {
    `...`
  }```

**Comment by rymai** [comment]
That's because the hooks/callbacks system is very flexible and allows a plugin to call `hook "xxxxxxx"` and that would call your `:xxxxxxx` callback.

That being said, `callback(:run_all_end)` should work *for plugins that define a `#run_all` method*.

**Comment by dogweather** [comment]
> That being said, callback(:run_all_end) should work for plugins that define a #run_all method.

I'm skeptical about that aspect of the design too :-) It means that a user of a plugin must familiarize themselves with its implementation details.

**Comment by rymai** [comment]
> I'm skeptical about that aspect of the design too :-) It means that a user of a plugin must familiarize themselves with its implementation details.

Yes that's the constraint for built-in hooks (https://github.com/guard/guard/wiki/Hooks-and-callbacks), but a plugin can also define custom hooks and document them, without the need for the user to know the implementation details (https://github.com/guard/guard/wiki/Hooks-and-callbacks#developers).
